core/coreapi/unixfs.go

```diff
@@ -183,7 +200,7 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.Node, opts ...options
 		if err != nil {
 			return path.ImmutablePath{}, err
 		}
-		mr, err := mfs.NewRoot(ctx, md, emptyDirNode, nil)
+		mr, err := mfs.NewRoot(ctx, md, emptyDirNode, nil, nil)
```

Comment:
new `nil` is the provider, which we leave disabled for MFS here. We are using a providing DAG service when needed.

Could we potentially passed a provider around and initialize MFS with it here and have it provide this way, rather than wrapping the DAGService? Potentially yes, but I'm not sure what happens with file-chunks and if MFS touches them at all. If not, I would need to wire the chunkers too etc. Overall it seemed more complex and fragile to do it that way.